eng/common/scripts/Generate-DocIndex.ps1

```diff
@@ -0,0 +1,147 @@
+        return $bomAwareString.Substring(3)
+    }
+    return $bomAwareString
+}
+Write-Verbose "Name Reccuring paths with variable names"
+$DocFxTool = "${RepoRoot}/docfx/docfx.exe"
+$DocOutDir = "${RepoRoot}/docfx_project"
+
+Write-Verbose "Initializing Default DocFx Site..."
+& "${DocFxTool}" init -q -o "${DocOutDir}"
+
+Write-Verbose "Copying template and configuration..."
+New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
+Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
+Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
+$YmlPath = "${DocOutDir}/api"
+New-Item -Path $YmlPath -Name "toc.yml" -Force
+
+Write-Verbose "Reading artifact from storage blob ..."
+$metadata = GetMetaData -lang $lang
+$langRegex = $LangMapping[$lang][1]
+$titleRegex = $LangMapping[$lang][0]
+$blobDirectoryRegex = "^$langRegex/(.*)/$"
+$pageToken = ""
+# Used for sorting the toc display order
+$orderSet = @{}
+$defaultGroupId = "com.azure"
+# This is a pagnation call as storage only return 5000 results as maximum.
+Do {
+    $resp = ""
+    if (!$pageToken) {
+        # First page call.
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex%2F&delimiter=%2F"
+    }
+    else {
+        # Next page call
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex%2F&marker=$pageToken"
+    }
+    # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+    
+    # Convert to xml documents.
+    $xmlDoc = [xml](removeBomFromString $resp)
+    foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+        # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+        $artifact = $elem.Name -replace $blobDirectoryRegex, '$1'
+        if ($blobPath -and ($blobPath -ne $artifact)) {
+            # $blobRentionArray = $blobPath -split "," | ForEach-Object { $_.Trim() }   
+            continue
+        }
+        # Some languages need to convert the artifact name, e.g azure-data-appconfiguration -> @azure/data-appconfiguration
+        if ($packageNameRegex) {
+            $artifact = $artifact -replace $packageNameRegex, $packageNameReplacement
+        }
+        # Read the artifact package infomation from csv of Azure/azure-sdk/_data/release/latest repo.
+        $packagesInfo = $metadata | ? { $_.Package -eq $artifact -and ($_.GroupId -eq $defaultGroupId)}
+        if ($packagesInfo.length -lt 1) {
+            $packagesInfo = $metadata | ? { $_.Package -eq $artifact }
+        }
+        if (!$packagesInfo) {
+            Write-Error "Did not find the artifacts from release csv. Please check and update."
```

Comment:
For the first time a package is release it will not have an entry in the metadata yet, so we might not want to cause an error but instead a warning.  I thought you were doing a warning and grouping in "other" at some point.